### PR TITLE
Make lightly.api independent of torch(vision) (breaking)

### PR DESCRIPTION
**Pull request by philippmwirth**
# Make lightly.api independent of torch(vision) (breaking)

### What changed?

This pull-request includes the following three (already reviewed) pull-requests:
* Drop all imports in `lightly.utils.__init__.py`. (https://github.com/lightly-ai/lightly/pull/1065)
* Drop all imports in `lightly/__init__.py`. (https://github.com/lightly-ai/lightly/pull/1066)
* Remove references to `LightlyDataset` in the `ApiWorkflowClient` (#1067)

Additionally, I had to make some additional small changes:
- https://github.com/lightly-ai/lightly/pull/1069/commits/560e2541825cc50bffa70b2539ea16e980d8c75c
- https://github.com/lightly-ai/lightly/pull/1069/commits/2aa3c4f02f43ce922b2f9e6d4bdc936584341655

Note: This breaks the way our imports work. For example:
```python
# this no longer works:
import lightly
lightly.api.ApiWorkflowClient

# use this instead
from lightly.api import ApiWorkflowClient

# or this
from lightly import api
api.ApiWorkflowClient
```

### How was it tested?
```bash
# with everything installed ordinarily
python
>>> from lightly import api
>>> exit()
# works

pip uninstall torch
python
>>> from lightly import api
>>> exit()
# works

pip uninstall torchvision
python
>>> from lightly import api
>>> exit()
# works

# setup with incompatible torch and torchvision (not compiled for same cuda)
python
>>> from lightly import api
>>> exit()
# works
```

### CODE REVIEW

File: lightly/__init__.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def _is_prefetch_generator_available():
         _torchvision_vit_available = True
     except (
         ImportError,          # No installation or old version of torchvision
-        OSError,              # Different CUDA versions for torch and torchvision
+        RuntimeError,         # Different CUDA versions for torch and torchvision
```
Comment (philippmwirth) [inline lightly/__init__.py:114]: `torchvision` raises a `RuntimeError` not a `OSError`.

File: lightly/api/api_workflow_upload_dataset.py [hunk 2]
```diff
@@ -39,7 +39,7 @@
     from lightly.data import LightlyDataset
     _lightly_dataset_available = True
 except (
-    OSError,      # Different CUDA versions for torch and torchvision
+    RuntimeError,         # Different CUDA versions for torch and torchvision
```
Comment (philippmwirth) [inline lightly/api/api_workflow_upload_dataset.py:42]: `torchvision` raises a `RuntimeError` not a `OSError`.

File: lightly/api/api_workflow_download_dataset.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 from PIL import Image
 
 from lightly.api.utils import paginate_endpoint, retry
-from torch.utils.hipify.hipify_python import bcolors
+from lightly.utils.hipify import bcolors
```
Comment (philippmwirth) [inline lightly/api/api_workflow_download_dataset.py:10]: Use our own hipify 😎

File: lightly/api/api_workflow_upload_dataset.py (hunk 2 above)
Comment (diksha) [reply lightly/api/api_workflow_upload_dataset.py:42]:
```
OSError: libcudnn.so.8: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/pip_deps_sematic/site-packages/sematic/resolvers/worker.py", line 159, in main
    func: Calculator = run.get_func()  # type: ignore
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/pip_deps_sematic/site-packages/sematic/db/models/run.py", line 231, in get_func
    func = getattr(importlib.import_module(import_path), func_name)
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/python_x86_64-unknown-linux-gnu/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/voxel/core/ml/data/curation/lib/lightly_worker.py", line 31, in <module>
    from lightly.api import ApiWorkflowClient
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/pip_deps_lightly/site-packages/lightly/__init__.py", line 110, in <module>
    import torchvision.models.vision_transformer
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/pip_deps_torchvision/site-packages/torchvision/__init__.py", line 4, in <module>
    import torch
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/pip_deps_torch/site-packages/torch/__init__.py", line 217, in <module>
    _load_global_deps()
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/pip_deps_torch/site-packages/torch/__init__.py", line 177, in _load_global_deps
    raise err
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/pip_deps_torch/site-packages/torch/__init__.py", line 172, in _load_global_deps
    ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
  File "/app/core/infra/sematic/perception/data_ingestion/door_data/main_lightly-image_image.binary.runfiles/python_x86_64-unknown-linux-gnu/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libcudnn.so.8: cannot open shared object file: No such file or directory
```
This is stacktrace of what we are seeing. Failing even with this

Comment (philippmwirth) [reply lightly/api/api_workflow_upload_dataset.py:42]: Thank you, @diksha. I'll include the `OSError` again.

**Review by guarin — CHANGES_REQUESTED** [review]
I believe there are still `torch.utils.hipify` imports in the `CLI` files (`crop_cli.py`, `embed_cli.py`, `train_cli.py`, `upload_cli.py`).

**Review by guarin — APPROVED** [review]

**Pull request merged**